typespec/Applications.Core/environments.tsp

```diff
@@ -65,11 +65,44 @@ model EnvironmentProperties {
   @doc("Specifies Recipes linked to the Environment.")
   recipes?: Record<Record<RecipeProperties>>;
 
+  @doc("Specifies configurations needed for the recipes registered to the environment.")
+  recipeConfig?: RecipeConfigProperties;
+
   @doc("The environment extension.")
   @extension("x-ms-identifiers", [])
   extensions?: Array<Extension>;
 }
 
+@doc("Specifies recipe configurations needed for the recipes.")
+model RecipeConfigProperties {
+  @doc("Specifies the terraform config properties")
+  terraform?: TerraformConfigProperties;
+}
+
+@doc("Specifies the terraform config properties")
+model TerraformConfigProperties{
+  @doc("Specifies authentication information needed to access private terraform module sources. Currently supported module sources: Git.")
+  authentication?: AuthConfig;
+}
+
+@doc("Specifies authentication information needed to access private terraform module repositories.")
+model AuthConfig{
+  @doc("Specifies authentication information needed to access private terraform modules from Git repository sources.")
+  git?: GitAuthConfig;
+}
+
+@doc("Specifies authentication information needed to access private terraform modules from Git repository sources.")
+model GitAuthConfig{
+  @doc("Specifies the secret details of type personal access token for each different git platforms")  
+  pat?: Record<SecretConfig>;
+}
+
+@doc("Specifies the secret details of type personal access token for each different git platforms")
+model SecretConfig {
+  @doc("The resource id for the Applications.Core/SecretStore resource containing credentials.")
```

Comment:
Could you add some information about what the secret name is required to be in the secret store? Do we expect that there needs to be a secret with a specific name?